bcdaworker/worker/worker.go

```diff
@@ -70,28 +72,28 @@ func (w *worker) ProcessJob(ctx context.Context, job models.Job, jobArgs models.
 
 	aco, err := w.r.GetACOByUUID(ctx, job.ACOID)
 	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("ProcessJob: could not retrieve ACO from database by UUID %s", job.ACOID))
+		err = errors.Wrap(err, fmt.Sprintf("ProcessJob: could not retrieve ACO from database by UUID %s", job.ACOID))
+		log.Worker.WithFields(logFields).Error(err)
+		return err
 	}
 
-	log.Worker.WithFields(logrus.Fields{
-		"jobID": jobArgs.ID,
-		"cmsID": *aco.CMSID,
-	})
+	logFields["cms_id"] = aco.CMSID
+	ctx = workerlog.WithLogFields(ctx, logFields)
 
 	err = w.r.UpdateJobStatusCheckStatus(ctx, job.ID, models.JobStatusPending, models.JobStatusInProgress)
 	if goerrors.Is(err, repository.ErrJobNotUpdated) {
 		// could also occur if job was marked as cancelled
-		log.Worker.Warnf("Failed to update job. Assume job already updated. Continuing. %s", err.Error())
+		log.Worker.WithFields(logFields).Warnf("Failed to update job. Assume job already updated. Continuing. %s", err.Error())
```

Comment:
I think this is where it starts to be a little unfortunate to not have the logger housed directly in the context and updated with the fields as needed 😅 It's very easy to forget to insert the logFields when logging a message.

Reply:
We could set it up similar to here, where we store an Entry. As new fields are added, we would need to add those to the Entry in the context, using the `WithField(s)` function. We could adjust this logic so that when a new field is added to the log Entry, instead of including the WithFields(logFields) line each time, we could update the Entry in the context and return it immediately for use again. 

If we add, say, two new fields with a function call, we would still need to immediately update the context but that would eliminate the need to include the new fields in each log call.

 What do you think?

Reply:
A potential downside is that the `Entry` being returned will no longer be referred to as `log.API...`, but whatever new name we decide to give to it. It could still use the `log.API` logger as the [Entrys Logger](https://pkg.go.dev/github.com/sirupsen/logrus#Entry).